airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/DatabaseConfigPersistence.java

```diff
@@ -1859,7 +1859,19 @@ void updateConfigsFromSeed(final DSLContext ctx, final ConfigPersistence seedCon
 
       // Process connector in use
       if (connectorRepositoriesInUse.contains(repository)) {
-        if (newFields.size() == 0) {
+        final String latestImageTag = latestDefinition.get("dockerImageTag").asText();
+        if (hasNewPatchVersion(connectorInfo.dockerImageTag, latestImageTag)) {
+          // Update connector to the latest patch version
+          LOGGER.info("Connector {} needs update: {} vs {}", repository, connectorInfo.dockerImageTag, latestImageTag);
+          if (configType == ConfigSchema.STANDARD_SOURCE_DEFINITION) {
+            writeStandardSourceDefinition(Collections.singletonList(Jsons.object(latestDefinition, StandardSourceDefinition.class)), ctx);
+          } else if (configType == ConfigSchema.STANDARD_DESTINATION_DEFINITION) {
+            writeStandardDestinationDefinition(Collections.singletonList(Jsons.object(latestDefinition, StandardDestinationDefinition.class)), ctx);
+          } else {
+            throw new RuntimeException("Unknown config type " + configType);
+          }
```

Comment:
nice! is this code the same as what's on lines 1908-1915. can we refactor it into a helper method? 

I actually think depending on how you write the helper, you could probably refactor 1895-1903. I think it is the same code with a different definition passed in as an arg.

Reply:
it's very detail oriented code and so it is risky to have it repeated in 3 places.